File: src/main/java/com/Backend/music_app/dto/request/AlbumCreateRequest.java
package com.Backend.music_app.dto.request;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AccessLevel;
import lombok.Data;
import lombok.experimental.FieldDefaults;

import java.time.LocalDate;
import java.util.UUID;

@Data
@FieldDefaults(level = AccessLevel.PRIVATE)
public class AlbumCreateRequest {

    @NotBlank(message = "Tên album không được để trống")
    String title;

    @NotNull(message = "ID ca sĩ không được để trống")
    UUID artistId;

    String coverUrl;

    LocalDate releaseDate;

    String description;
}

File: lib/data/createRequest/AlbumCreate_request.dart
import 'package:flutter/cupertino.dart';

class AlbumCreateRequest{
  final String title;
  final String artistId;
  final String coverUrl;
  final DateTime releaseDate;
  final String description;

  AlbumCreateRequest({
    required this.title,
    required this.artistId,
    required this.coverUrl,
    required this.releaseDate,
    required this.description,
  });

  Map<String, dynamic> toJson() {
    return {
      'title': title,
      'artistId': artistId,
      'coverUrl': coverUrl,
      'releaseDate': releaseDate.toIso8601String(),
      'description': description,
    };
  }
}